src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -282,7 +283,14 @@ class MarathonSchedulerActor private (
 
   def deploymentFailed(plan: DeploymentPlan, reason: Throwable): Unit = {
     logger.error(s"Deployment ${plan.id}:${plan.version} of ${plan.targetIdsString} failed", reason)
-    Future.sequence(plan.affectedRunSpecIds.map(launchQueue.purge))
+    Future.sequence(plan.affectedRunSpecIds.map { id =>
+      instanceTracker.specInstances(id).map { instances =>
+        Future.sequence(instances.map { instance =>
```

Comment:
As @timcharper pointed out, I think this part wouldn't work before we merge the refactoring from Matthias. I believe that nobody would kill those instances. Before it worked because we force expunged.